OrbitGl/App.cpp

```diff
@@ -1523,22 +1524,36 @@ void OrbitApp::DeselectTracepoint(const TracepointInfo& tracepoint) {
 }
 
 void OrbitApp::AddFrameTrack(const FunctionInfo& function) {
-  GetMutableCaptureData().InsertFrameTrack(function);
-  data_manager_->set_user_defined_capture_data(GetCaptureData().user_defined_capture_data());
-  AddFrameTrackTimers(function);
+  data_manager_->InsertFrameTrack(function);
+
+  if (!HasCaptureData()) {
+    return;
+  }
+  // We only add a frame track to the actual capture data if the function for the frame
+  // track actually has hits in the capture data. Otherwise we can end up in inconsistent
+  // states where "empty" frame tracks exist in the capture data (which would also be
+  // serialized).
+  const FunctionStats& stats = GetCaptureData().GetFunctionStatsOrDefault(function);
```

Comment:
That makes very much sense. Maybe we should add a pop-up here, explaining this to the user (by here, I really mean after the !HasCaptureData.

Reply:
Added pop-up that has checkbox option to not show again.